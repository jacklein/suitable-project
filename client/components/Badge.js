import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, StyleSheet, Dimensions, Image } from 'react-native'
import { Card, Button } from 'react-native-elements';
import { Bar } from 'react-native-progress';
import { primaryColor } from '../styles/common';

const SCREEN_WIDTH = Dimensions.get('window').width;

class Badge extends React.PureComponent {
  render() {
    const { details } = this.props.badge;
    return (
      <View style={styles.container}>
        <Card containerStyle={styles.card}>
          <View style={{ flexDirection: 'row' }}>
            <Image
              style={styles.image}
              source={{ uri: details.iconUrl }}
            />
            <View style={{ flex: 1, marginTop: 10 }}>
              <Text style={styles.title}>
                {details.title}
              </Text>
              <Bar 
                progress={details.progress}
                color={primaryColor}
                height={15}
              />
            </View>
          </View>
          <Text style={styles.description}>
            {details.description}
          </Text>
          <Button
            backgroundColor={primaryColor}
            buttonStyle={styles.button}
            title='View Tasks'
            onPress={() => this.props.onPress(this.props.badge)}
          />
        </Card>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center'
  },
  card: {
    borderWidth: 0,
    width: SCREEN_WIDTH, 
    marginTop: 0, 
    marginBottom: 10
  },
  image: {
    left: -15,
    resizeMode: 'contain',
    width: 100,
    height: 100
  },
  title: {
    marginBottom: 10,
    fontWeight: 'bold'
  },
  description: {
    marginBottom: 20, 
    marginTop: 15
  },
  button: {
    borderRadius: 0,
    alignSelf: 'center'
  }
});

Badge.propTypes = {
  badge: PropTypes.object,
  onPress: PropTypes.func
}

export default Badge;