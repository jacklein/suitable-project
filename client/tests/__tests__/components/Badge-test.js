import React from 'react';
import { shallow } from 'enzyme';
import Badge from '../../../components/Badge';
import { achievements } from '../../mockdata';

describe('Badge component', () => {
  test('renders correctly', () => {
    const wrapper = shallow(
      <Badge badge={achievements[0]} index={0} onPress={() => {}}/>
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({ index: 1 });
    expect(wrapper).toMatchSnapshot();
  });
});