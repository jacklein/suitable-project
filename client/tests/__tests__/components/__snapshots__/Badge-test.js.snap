// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge component renders correctly 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Card
    containerStyle={
      Object {
        "borderWidth": 0,
        "marginBottom": 10,
        "marginTop": 0,
        "paddingTop": 0,
        "width": 750,
      }
    }
  >
    <Component
      style={
        Object {
          "flexDirection": "row",
          "top": 5,
        }
      }
    >
      <Component
        source={
          Object {
            "uri": "https://www.filepicker.io/api/file/h6spFMWTQaKXzk6HOxqL",
          }
        }
        style={
          Object {
            "height": 242.54545454545453,
            "left": -15,
            "resizeMode": "contain",
            "width": 242.54545454545453,
          }
        }
      />
      <Component
        style={
          Object {
            "flex": 1,
            "marginTop": 10,
          }
        }
      >
        <Component
          style={
            Object {
              "fontWeight": "bold",
              "marginBottom": 10,
            }
          }
        >
          Exceptional Achievement
        </Component>
        <ProgressBar
          animated={true}
          animationConfig={
            Object {
              "bounciness": 0,
            }
          }
          animationType="spring"
          borderRadius={4}
          borderWidth={1}
          color="#4492E5"
          height={15}
          indeterminate={false}
          progress={0.29699748744715526}
          useNativeDriver={false}
          width={150}
        />
      </Component>
    </Component>
    <Component
      style={
        Object {
          "marginBottom": 15,
          "marginTop": 10,
        }
      }
    >
      Sit nisi eveniet non expedita eos. Porro et odit vel quis voluptatem error. Et est autem nostrum quis voluptatem voluptas voluptatibus sint. Occaecati nesciunt provident dolorem tenetur rerum officiis.
    </Component>
    <Button
      backgroundColor="#4492E5"
      buttonStyle={
        Object {
          "alignSelf": "center",
          "borderRadius": 0,
          "marginBottom": 15,
        }
      }
      onPress={[Function]}
      title="View Activities"
    />
  </Card>
</Component>
`;

exports[`Badge component renders correctly 2`] = `
<Component
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Card
    containerStyle={
      Object {
        "borderWidth": 0,
        "marginBottom": 10,
        "marginTop": 0,
        "paddingTop": 0,
        "width": 750,
      }
    }
  >
    <Component
      style={
        Object {
          "flexDirection": "row",
          "top": 5,
        }
      }
    >
      <Component
        source={
          Object {
            "uri": "https://www.filepicker.io/api/file/h6spFMWTQaKXzk6HOxqL",
          }
        }
        style={
          Object {
            "height": 242.54545454545453,
            "left": -15,
            "resizeMode": "contain",
            "width": 242.54545454545453,
          }
        }
      />
      <Component
        style={
          Object {
            "flex": 1,
            "marginTop": 10,
          }
        }
      >
        <Component
          style={
            Object {
              "fontWeight": "bold",
              "marginBottom": 10,
            }
          }
        >
          Exceptional Achievement
        </Component>
        <ProgressBar
          animated={true}
          animationConfig={
            Object {
              "bounciness": 0,
            }
          }
          animationType="spring"
          borderRadius={4}
          borderWidth={1}
          color="#4492E5"
          height={15}
          indeterminate={false}
          progress={0.29699748744715526}
          useNativeDriver={false}
          width={150}
        />
      </Component>
    </Component>
    <Component
      style={
        Object {
          "marginBottom": 15,
          "marginTop": 10,
        }
      }
    >
      Sit nisi eveniet non expedita eos. Porro et odit vel quis voluptatem error. Et est autem nostrum quis voluptatem voluptas voluptatibus sint. Occaecati nesciunt provident dolorem tenetur rerum officiis.
    </Component>
    <Button
      backgroundColor="#4492E5"
      buttonStyle={
        Object {
          "alignSelf": "center",
          "borderRadius": 0,
          "marginBottom": 15,
        }
      }
      onPress={[Function]}
      title="View Activities"
    />
  </Card>
</Component>
`;
